### DISCUSSION TITLE
本科生诚信承诺书文本中如何插入论文题目

### DISCUSSION THREAD

**Discussion by fantasyEthan** [nju-lug/NJUThesis#133] [Q&A]
<img width="489" alt="image" src="https://user-images.githubusercontent.com/54493995/163962123-8af7c854-c0d0-4428-a6d7-29d154b3426b.png">
如题

**Comment by note286** [comment]
应该是这个文件里的，`C:\texlive\2022\texmf-dist\tex\latex\njuthesis\njuthesis-undergraduate.def`，拷贝出来到你本地的主tex文件目录下，然后修改对应内容即可，没有实测，你试试，应该是可以的。

**Comment by atxy-blip** [comment]
~~感谢  @note286 解答，改动 `.def` 文件然后挂载是完全可行的。但之前给别人这么改的时候还是太麻烦，所以做了一些优化。~~

~~实际上 v0.17.0 预留了一个更方便的接口：~~

```TeX
\njusetname{origdecl}{你喜欢的名字}
```

-----

这个已经在 v0.17.0 修复了，新版本生成的本科承诺书自动附带你的论文题目。

![image](https://user-images.githubusercontent.com/63113546/164160067-75141a39-3176-4518-a414-0c48715c2199.png)


在这个版本中文字本身也能被修改：

```TeX
\njusettext{origdecl}{你喜欢的承诺文字}
```

**Comment by fantasyEthan** [comment]
好的谢谢解答，因为我是在云端直接写没有本地编译，所以 @note286 的办法在我这里可能会麻烦一点。感谢作者 @atxy-blip 的方法。

**Reply by atxy-blip** [reply]
较真来讲，由于 <https://tex.nju.edu.cn> 还没更新 TeX Live 2022（咕），内置的是 `v0.16.0`，实际上你也需要上传 `.cls` `.def` 文件才能使用新版，第一种方法并不会产生额外的麻烦。但就分离模板和内容本身而言，我是不建议修改 `.cls` `.def` 文件的。

**Reply by atxy-blip** [reply]
啊，如果你说的是“题目：”冒号后面空着的情况，新版本应该也是解决了的。

**Reply by fantasyEthan** [reply]
哈哈哈哈哈是的，我刚才试了一下显示是undefined control sequence，那我就使用第一种方法吧哈哈哈，还是谢谢作者

**Reply by fantasyEthan** [reply]
对的，我说的是后者的情况

**Reply by fantasyEthan** [reply]
我刚才看了一下说明，我可能题目问的有点误导性，我的问题是承诺书中题目的问题。当然可以在打印签字之后手动 \includepdf加进去

**Reply by atxy-blip** [reply]
那你就下载 [njuthesis-user-v0.17.0.zip](https://github.com/nju-lug/NJUThesis/releases/download/v0.17.0/njuthesis-user-v0.17.0.zip)，先把里面的 `njuthesis.cls` 和 `njuthesis-undergraduate.def` 上传到在线项目里。

**Reply by fantasyEthan** [reply]
嗯嗯我明白你的意思了，谢谢！我想应该也会有别人有类似的疑问

**Reply by atxy-blip** [reply]
我修改了题目描述和回答，大概不会有误导性了。
